exp/controllers/azuremachinepoolmachine_controller_test.go

```diff
@@ -172,8 +172,9 @@ func getAReadyMachinePoolMachineCluster() (*clusterv1.Cluster, *infrav1.AzureClu
 
 	ampm := &infrav1exp.AzureMachinePoolMachine{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      "ampm1",
-			Namespace: "default",
+			Name:       "ampm1",
+			Namespace:  "default",
+			Finalizers: []string{"test"},
```

Comment:
What was the purpose of adding the finalizer here?

Reply:
Good question. I answered [above](https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/3707#discussion_r1289120353), but basically some tests using CAPI's fake client will now error without finalizers.